### PR TITLE
8344120: Remove Security Manager dependencies from jdk.crypto.cryptoki module

### PR DESCRIPTION
**Pull request by seanjmullan**
Now that JEP 486 has been integrated, the `jdk.crypto.cryptoki` implementation dependencies on `System.getSecurityManager` and `AccessController.doPrivileged` can be removed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344120](https://bugs.openjdk.org/browse/JDK-8344120): Remove Security Manager dependencies from jdk.crypto.cryptoki module (**Enhancement** - P3)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Anthony Scarpino](https://openjdk.org/census#ascarpino) (@ascarpino - **Reviewer**) Review applies to [f3259bdc](https://git.openjdk.org/jdk/pull/22077/files/f3259bdc2fdd312a7f1e9d86a67c06153b4996a6)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22077/head:pull/22077` \
`$ git checkout pull/22077`

Update a local copy of the PR: \
`$ git checkout pull/22077` \
`$ git pull https://git.openjdk.org/jdk.git pull/22077/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22077`

View PR using the GUI difftool: \
`$ git pr show -t 22077`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22077.diff">https://git.openjdk.org/jdk/pull/22077.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22077#issuecomment-2473839939)
</details>

### CODE REVIEW

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/Config.java [hunk 1]
```diff
@@ -62,27 +60,16 @@ final class Config {
...
+                "sun.security.pkcs11.allowSingleThreadedModules", "true");
+        if ("false".equalsIgnoreCase(allowSingleThreadedModules)) {
             staticAllowSingleThreadedModules = false;
         } else {
             staticAllowSingleThreadedModules = true;
         }
-        osName = props.get(1);
-        osArch = props.get(2);
+        osName = System.getProperty("os.name");
+        osArch = System.getProperty("os.arch");
```
Comment (RogerRiggs) [inline src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/Config.java:72]: osName and osArch seem to be unused; they could be removed but might be out of scope for this PR.

**Review by RogerRiggs — APPROVED** [review]
LGTM

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyAgreement.java [hunk 2]
```diff
@@ -73,11 +73,8 @@ private static class AllowKDF {
 
         private static final boolean VALUE = getValue();
 
-        @SuppressWarnings("removal")
         private static boolean getValue() {
-            return AccessController.doPrivileged(
-                (PrivilegedAction<Boolean>)
-                () -> Boolean.getBoolean("jdk.crypto.KeyAgreement.legacyKDF"));
+            return Boolean.getBoolean("jdk.crypto.KeyAgreement.legacyKDF");
```
Comment (ascarpino) [inline src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyAgreement.java:77]: I'm fine with the code as is, but is there a reason why `Boolean.getBoolean(...)` is not replacing the method call on line 74?  Is there initialization/startup concerns?

**Review by ascarpino — APPROVED** [review]
One minor question, but it looks good as is.

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/Config.java (hunk 1 above)
Comment (seanjmullan) [reply src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/Config.java]: Good spot; I removed them - I'm ok with making small changes as long as they are within the scope of the code being modified.

File: src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyAgreement.java (hunk 2 above)
Comment (seanjmullan) [reply src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyAgreement.java]: I removed the `getValue` method and changed VALUE to call `getBoolean`. It's also debatable whether it needs to be lazy initialized in a static class, but that is getting a bit beyond the scope of these changes so I left that as-is.

**Review by RogerRiggs — APPROVED** [review]
lgtm

**Pull request closed without merging**